src/lib/isLength.js

```diff
@@ -5,15 +5,41 @@ export default function isLength(str, options) {
   assertString(str);
   let min;
   let max;
+  let exact;
+  let result;
+  let isValid = false;
   if (typeof (options) === 'object') {
     min = options.min || 0;
     max = options.max;
+    exact = options.exact;
   } else { // backwards compatibility: isLength(str, min [, max])
     min = arguments[1] || 0;
     max = arguments[2];
   }
   const presentationSequences = str.match(/(\uFE0F|\uFE0E)/g) || [];
   const surrogatePairs = str.match(/[\uD800-\uDBFF][\uDC00-\uDFFF]/g) || [];
   const len = str.length - presentationSequences.length - surrogatePairs.length;
-  return len >= min && (typeof max === 'undefined' || len <= max);
+  result = len >= min && (typeof max === 'undefined' || len <= max);
+  if (result === false || typeof exact === 'undefined') return result;
```

Comment:
Why is it not the following?
```suggestion
  if (result === true && typeof exact === 'undefined') return result;
```

Reply:
And maybe we can refactor this so we use `isValid` instead of this additional `result` variable

Reply:
> Why is it not the following?

Firstly we check if the Length of the String falls in the range `[min, max]` and the result is stored in the `result` variable. Now, if the `result` is `false,` it means the Length is out of range `[min, max]` and we return the `result` without checking for the `exact` condition. Now if `result` stores `true` and if `exact` is not defined i.e. `exact` is not passed, then also we return `result`, but if `result` stores `true` and `exact` is defined, we do not return `result` since any of the conditions is not true. In that case, we check for `exact` conditions. 
I hope this solves the query...

Reply:
> And maybe we can refactor this so we use `isValid` instead of this additional `result` variable

As mentioned earlier, We check if the Length of the String falls in the range `[min, max]` and the result is stored in the `result` variable, So it would be far easier to just calculate the condition once and then store it in a variable and do all the operations on that variable instead of calculating it, again and again, whenever required.
No doubt we can do it without adding the  `result` variable but then I think it would increase the code repetition and complexity and indeed the overall response time.

Reply:
Ah yes, you are right. Thanks for the explanation!

Reply:
suggestion: for readability's sake, the variable `result` should probably be `isWithinRange` or `isInsideRange` 😄.

Also, a better technical term for `exact` would be `discreteLengths` (or perhaps you have a more meaningful or shorter term :P). `exact` sounds like a boolean and could be misunderstood. "Discrete", because you're comparing with one or more discrete numbers.

Also, I would highly recommend reducing the complexity of this solution by opting only for one data type for `exact`/`discreteLengths`, because simplicity is key :). Why do you require support for dictionary, if I may ask? Shouldn't Array be sufficient to cover all use cases?

You can simplify by only supporting Array. It reduced the API footprint and, therefore, complexity. Then you can also remove those tests.

I don't think you need more than the following:

```suggestion
const isInsideRange =
len >= min && (typeof max === 'undefined' || len <= max);

if (isInsideRange && Array.isArray(options?.discreteLengths)) {
    return options.discreteLengths.some(discreteLen => discreteLen === len);
}

return isInsideRange;
```

PS: I came to this code whilst looking for easy issues for Hacktoberfest and this was marked for that purpose, but already taken 😄. I still found it intriguing. That's why I checked the implementation. No offense :).